sharding/p2p/message.go

```diff
@@ -0,0 +1,9 @@
+package p2p
+
+// Message represents a message received from an external peer.
+type Message struct {
+	// Peer represents the sender of the message.
+	Peer Peer
+	// Data can be any type of message found in sharding/p2p/messages package.
```

Comment:
Talk about type assertions here in comments and how individual services can verify a message is of a certain type by something along the lines of:

```go
// Type assertion helps us catch incorrect data requests.
msg, ok := req.Data.(sharding.CollationBodyRequest)
if !ok {
	log.Error(fmt.Sprintf("Received incorrect data request type: %v", msg))
}
```

Reply:
I disagree. This makes the godoc very busy. There is an example of using this message with the Feed and I think type assertion is fundamental basic knowledge of Go.